hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -52,10 +52,14 @@ private[hudi] object HoodieSparkSqlWriter {
 
     val sparkContext = sqlContext.sparkContext
     val path = parameters.get("path")
-    val tblName = parameters.get(HoodieWriteConfig.TABLE_NAME)
-    if (path.isEmpty || tblName.isEmpty) {
+    var tblName_tmp = parameters.get(HoodieWriteConfig.TABLE_NAME)
+
+    if ( !path.isEmpty && !tblName_tmp.isEmpty) {
+      val tblName = tblName_tmp.get.trim
+    }else{
       throw new HoodieException(s"'${HoodieWriteConfig.TABLE_NAME}', 'path' must be set.")
     }
+
```

Comment:
this would be replaced with 
```java
val tblNameOp = parameters.get(HoodieWriteConfig.TABLE_NAME)
    if (path.isEmpty || tblNameOp.isEmpty) {
      throw new HoodieException(s"'${HoodieWriteConfig.TABLE_NAME}', 'path' must be set.")
    }
    val tblName = tblNameOp.get.trim
```